src/include/compat.h

```diff
@@ -116,25 +116,29 @@
 
 #if defined(HAVE_PTHREAD_SETNAME_NP)
   #if defined(__APPLE__)
-    #define pthread_setname_np(thread, name) ({ \
+    #define ceph_pthread_setname(thread, name) ({ \
       int __result = 0;                         \
       if (thread == pthread_self())             \
-        __result = pthread_setname_np(name)     \
+        __result = pthread_setname_np(name);    \
```

Comment:
@ukernel

> Defining a two parameters version pthread_setname_np() does not work.

@dx9 has addressed this issue, what fails the the osx build is the ";", i think. so no need to define "ceph_pthread_{set,get}name()" right?